lib/segment/src/vector_storage/dynamic_mmap_flags.rs

```diff
@@ -0,0 +1,207 @@
+use std::cmp::max;
+use std::fs::create_dir_all;
+use std::path::{Path, PathBuf};
+
+use bitvec::prelude::BitSlice;
+use memmap2::MmapMut;
+
+use crate::common::error_logging::LogError;
+use crate::common::mmap_ops::{
+    create_and_ensure_length, open_write_mmap, read_from_mmap, write_to_mmap,
+};
+use crate::entry::entry_point::{OperationError, OperationResult};
+use crate::vector_storage::chunked_utils::{ensure_status_file, MmapStatus};
+use crate::vector_storage::div_ceil;
+use crate::vector_storage::mmap_vectors::mmap_to_bitslice;
+
+#[cfg(test)]
+const MINIMAL_MMAP_SIZE: usize = 1024; // 1Kb
+
+#[cfg(not(test))]
+const MINIMAL_MMAP_SIZE: usize = 1024 * 1024; // 1Mb
+
+const FLAGS_FILE: &str = "flags.dat";
+
+pub struct DynamicMmapFlags {
+    status: MmapStatus,
+    _flags_mmap: Option<MmapMut>,
+    flags: Option<&'static mut BitSlice>,
+    status_mmap: MmapMut,
```

Comment:
Do we need memmap for `status_mmap`? As I understand, this file contains only the points count. Can we use serde instead?

Reply:
don't want to open file each time. mmap should be faster

Reply:
Is it possible to update file in the `flush` function only? I guess it's a correct scenario